staging/src/k8s.io/apimachinery/pkg/util/httpstream/spdy/roundtripper_test.go

```diff
@@ -400,16 +410,27 @@ func TestRoundTripAndNewConnection(t *testing.T) {
 				}
 			}
 
-			var expectedProxyAuth string
+			var auth string
+			var authHeader string
 			if testCase.proxyAuth != nil {
-				encodedCredentials := base64.StdEncoding.EncodeToString([]byte(testCase.proxyAuth.String()))
-				expectedProxyAuth = "Basic " + encodedCredentials
+				username := testCase.proxyAuth.Username()
+				password, _ := testCase.proxyAuth.Password()
+				auth = username + ":" + password
+				authHeader = "Basic " + base64.StdEncoding.EncodeToString([]byte(auth))
```

Comment:
keep the name of this `expectedProxyAuth` to make it clear which is the data in the testcase and which is the data we're expecting

both the old test code and new test code share a flaw of essentially duplicating the implementation, so if the implementation is wrong, our copy in the test will be wrong too, and the test will still pass

can we improve this to actually extract the username and password from the header using a stdlib method and see if the username and password match the testcase data? (example at https://go.dev/play/p/psQsaWVA4ZG)

Reply:
it's now expectedUsername and expectedPassword, thanks for the suggestion. although it doesn't get rid of all the duplication of logic, at least the base64 decoding is now being handled by stdlib.